src/plonetheme/eeq/interfaces.py

```diff
@@ -2,7 +2,45 @@
 """Module where all interfaces, events and exceptions live."""
 
 from zope.publisher.interfaces.browser import IDefaultBrowserLayer
+from plone.supermodel import model
+from plone.app.textfield import RichText
+from zope import schema
 
 
 class IPlonethemeEeqLayer(IDefaultBrowserLayer):
     """Marker interface that defines a browser layer."""
+
+class ISettings(model.Schema):
+    top_banner_body = schema.Text(
+        title="Banner to display at the top of the site",
+        description="You can use the placeholder <pre>${portal_url}</pre> to refer to assets",
+        default="""
+            <div class="container-fluid constrain-width banner">
+                <div class="row py-3 banner-content">
+                    <div class="col-xs-12 col-md-10 col-lg-8 offset-md-1 offset-lg-2">
```

Comment:
Maybe this wrapper stuff belongs in the viewlet code rather than the configurable HTML?

Reply:
Moving the wrapper to the theme html has the advantage of providing the user with convenience boilerplate, but it requires a bit more complexity.

At least to maintain the current behaviour, i.e. copy the banner verbatim to the top of the theme html if it's found in the content html. Moving the wrapper to the theme would mean either always leaving it there or conditionally removing it when the banner is not present.

Among these two approaches:
* what you put in the control panel field will end up verbatim in the theme, right at the top
* we'll wrap your HTML in proper classes to make a banner

the latter initially sounds enticing, but the former is the one I'd prefer.
Even from the user perspective, it's simpler to understand what's going on.
If I don't sound conving enough I'll go and move the wrapper though.

Reply:
Another option would be to include the wrapper markup in the theme html, rather than the viewlet html. The rules file could drop it from the theme if the viewlet wasn't present in the content (e.g. `<drop css:theme="#above-header-banner" css:if-content="#above-header-content" />`). Either way the question is more of a UX one than a technical one, so perhaps we should defer to the client and/or @skleinfeldt on it.

For now, I think we should deploy this as you prefer it and they can play around with it and let us know how it works for them.